walkeeper/src/send_wal.rs

```diff
@@ -114,42 +72,39 @@ impl SendWalConn {
         let tli_bytes = tli.as_bytes();
         let sysid_bytes = sysid.as_bytes();
 
-        let mut outbuf = BytesMut::new();
-        BeMessage::write(
-            &mut outbuf,
-            &BeMessage::RowDescription(&[
-                RowDescriptor {
-                    name: b"systemid\0",
-                    typoid: 25,
-                    typlen: -1,
-                },
-                RowDescriptor {
-                    name: b"timeline\0",
-                    typoid: 23,
-                    typlen: 4,
-                },
-                RowDescriptor {
-                    name: b"xlogpos\0",
-                    typoid: 25,
-                    typlen: -1,
-                },
-                RowDescriptor {
-                    name: b"dbname\0",
-                    typoid: 25,
-                    typlen: -1,
-                },
-            ]),
-        );
-        BeMessage::write(
-            &mut outbuf,
-            &BeMessage::DataRow(&[Some(sysid_bytes), Some(tli_bytes), Some(lsn_bytes), None]),
-        );
-        BeMessage::write(
-            &mut outbuf,
-            &BeMessage::CommandComplete(b"IDENTIFY_SYSTEM\0"),
-        );
-        BeMessage::write(&mut outbuf, &BeMessage::ReadyForQuery);
-        self.stream_out.write_all(&outbuf)?;
+        pgb.write_message_noflush(&BeMessage::RowDescription(&[
+            RowDescriptor {
+                name: b"systemid\0",
+                typoid: 25,
+                typlen: -1,
+                ..Default::default()
+            },
+            RowDescriptor {
+                name: b"timeline\0",
+                typoid: 23,
+                typlen: 4,
+                ..Default::default()
+            },
+            RowDescriptor {
+                name: b"xlogpos\0",
+                typoid: 25,
+                typlen: -1,
+                ..Default::default()
+            },
+            RowDescriptor {
+                name: b"dbname\0",
+                typoid: 25,
+                typlen: -1,
+                ..Default::default()
+            },
+        ]))?;
+        pgb.write_message_noflush(&BeMessage::DataRow(&[
+            Some(sysid_bytes),
+            Some(tli_bytes),
+            Some(lsn_bytes),
+            None,
+        ]))?;
+        pgb.write_message_noflush(&BeMessage::CommandComplete(b"IDENTIFY_SYSTEM\0"))?;
```

Comment:
Should we flush here

Reply:
No, following ReadyForQuery will do that. Added comment to `process_query` about that.